docs/guides/data-movement/copy-offload.md

```diff
@@ -0,0 +1,92 @@
+```
+
+> [!IMPORTANT]
+> Copy the certificate to `/etc/nnf-dm-usercontainer/cert.pem` on each compute node. It must be readable by all users' compute applications.
+
+### Rabbit hostname in /etc/hosts
+
+Each compute node must know the name of its matching rabbit node. This allows the `libcopyoffload` library, used in the user's compute application, to know which rabbit has the copy-offload server for that job. The library expects to use a generic name, `local-rabbit`, to contact that rabbit.
+
+### WLM and the per-Workflow token
+
+The WLM, such as Flux, must retrieve the per-Workflow token and make it available to the user's compute application as an environment variable named `DW_WORKFLOW_TOKEN`. The token is used by the `libcopyoffload` library to construct the "Bearer Token" for its requests to the copy-offload server. The token becomes invalid after the Workflow enters its teardown state.
+
+The Workflow contains a reference to the name of the Secret that holds the token. The following value returns the name and namespace of the secret:
+
+```console
+kubectl get workflow $WORKFLOW_NAME -o json | jq -rM '.status.workflowToken'
+```
+
+If information about the token's secret is returned, then read the token from the given secret:
+
+```console
+TOKEN=$(kubectl get secret -n $SECRET_NAMESPACE $SECRET_NAME -o json | jq -rM '.data.token' | base64 -d)
+```
+
+Create the environment variable for the user's compute application:
+
+```bash
+DW_WORKFLOW_TOKEN="$TOKEN"
+```
+
+> [!NOTE]
+> Per-Workflow tokens are not limited to the copy-offload API. Any user container may request to be configured with the job's per-Workflow token and the TLS certificate. See `requires=user-container-auth` in [User Containers](../user-containers/readme.md). The WLM must always check for the existence of a token secret in the Workflow.
+
+## User Enablement of Copy Offload
+
+Users enable the copy-offload server by requesting it in their job script. The script must contain a `DW container` directive that specifies the desired copy-offload container profile. At least one of the `DW jobdw` or `DW persistentdw` directives in the job script must include the `requires=copy-offload` statement. See [User Interactions](../user-interactions/readme.md) for more details about these directives.
+
+The user's compute application must be linked with the `libcopyoffload` library. This library understands how to find and use the TLS certificate and the per-Workflow token required for communication with the copy-offload server for the user's job.
+
+The copy-offload server and its container profile are specified in the `container` directive. See [User Containers](../user-containers/readme.md) for details about container profiles. The following selects the default copy-offload container profile:
+
+```bash
+#DW jobdw name=my-job-name requires=copy-offload [...]
+#DW container name=copyoff-container profile=copy-offload-default [...]
```

Comment:
This is probably getting into the details of user containers, but my guess is this will be most users first introduction to using container directives. We should probably include the storage parameters for the container directive as an example since those need to also be present in the container profile and they need to point to the jobdw storage.

Reply:
I don't want to duplicate the user container doc here. I can look at the paragraph preceding this and come up with a better way to say "hey, you and your admin need to know that stuff in the user container doc for ....reasons".